src/routes/embedded/auth/recover-account/keyfile/auth-recover-account-keyfile.view.tsx

```diff
@@ -1,131 +1,136 @@
+    isLoading: isUploading,
+    error: uploadError,
+    importedWalletAddress,
+    parse: parseUpload,
+    reset: resetUpload,
+  } = useWalletUpload({
+    wallets: [],
+    importTempWallet,
+    allowRecoveryFile: false,
+    mustWalletExist: false,
+  });
+
+  const areButtonsDisabled = isRecovering || isUploading;
+  const isViewLoading = isUploading;
 
   const handleRecover = async () => {
     try {
-      setLoading(true);
+      if (areButtonsDisabled) return;
+
+      if (uploadError) {
+        toast.error(uploadError);
+
+        return;
+      }
+
+      if (!WalletUtils.isJWK(uploadData)) {
+        toast.error("Invalid file.");
+
+        return;
+      }
+
+      setIsRecovering(true);
+
       const recoverableAccounts = await fetchRecoverableAccounts();
+
       if (recoverableAccounts.length === 1) {
         await fetchRecoverableAccountWallets(recoverableAccounts[0]);
         navigate(EmbeddedPaths.Auth);
       } else if (recoverableAccounts.length > 1) {
         navigate("/auth/recover-account/select");
       } else {
         toast.error("No recoverable accounts found");
+        setIsRecovering(false);
       }
-      setLoading(false);
     } catch (error) {
-      toast.error(error?.message || "Error recovering account");
-      setLoading(false);
+      console.error(error);
+      toast.error("Unexpected error while recovering account.");
+      setIsRecovering(false);
     }
   };
 
+  const handleTryAgain = () => {
+    resetUpload();
+    deleteImportedTempWallet();
+  };
+
   useEffect(() => {
     deleteImportedTempWallet();
     clearRecoverableAccounts();
   }, []);
 
-  return importedTempWalletAddress ? (
-    <Card
+  return importedWalletAddress ? (
+    <OnboardingCard
       headerText="Import Keyfile"
       subtitle="Upload your private key to recover your account."
-      footerElement={<WanderFooter />}
-      hasBackButton={true}
-      onBackButtonClick={back}
-      hasCloseButton={true}
-      onCloseButtonClick={() => navigate(`/auth`)}
-      style={{ gap: 24 }}
-      size="auto">
+      onBackButtonClick={() => navigate(`/auth/recover-account`)}
+      isLoading={ isViewLoading }>
       <Text>Would you like to recover this account?</Text>
       <Copyable
         isFullWidth
         style={{ padding: 0 }}
         label="Your wallet address"
         onClick={() => {
-          copy(importedTempWalletAddress);
+          copy(importedWalletAddress);
         }}
-        value={importedTempWalletAddress}
+        value={importedWalletAddress}
       />
       <Row>
-        <Button variant="secondary" size="md" onClick={deleteImportedTempWallet}>
+        <Button variant="secondary" size="md" onClick={handleTryAgain} isDisabled={areButtonsDisabled}>
           No, try again
         </Button>
-        <Button variant="primary" size="md" onClick={handleRecover} isLoading={loading}>
+        <Button variant="primary" size="md" onClick={handleRecover} isDisabled={areButtonsDisabled}>
```

Comment:
There should be a loading either on this button or the screen itself when clicking on `Yes, recover` but is missing.